annif/corpus/document.py

```diff
@@ -60,13 +63,26 @@ def opener(path):
                 return gzip.open(path, mode='rt')
         else:
             opener = open
-
-        with opener(self.path) as tsvfile:
-            for line in tsvfile:
-                text, uris = line.split('\t', maxsplit=1)
-                subjects = [annif.util.cleanup_uri(uri)
-                            for uri in uris.split()]
-                yield Document(text=text, uris=subjects, labels=[])
+        try:
+            with opener(self.path) as tsvfile:
+                for line in tsvfile:
+                    yield from self._parse_tsv_line(line)
+        except FileNotFoundError as err:
+            raise AnnifException(str(err))
+
+    @staticmethod
+    def _parse_tsv_line(line):
+        try:
+            text, uris = line.split('\t', maxsplit=1)
+            subjects = [annif.util.cleanup_uri(uri)
+                        for uri in uris.split()]
+            yield Document(text=text, uris=subjects, labels=[])
+        except ValueError as err:
```

Comment:
Two things I dislike in this `try..except` block:

1. It contains 3 lines of code, involving several calls to external code, but the only actually covered problem happens on the first line. The two other lines should be moved outside the block.
2. Testing the specific type of exception using `'not enough values to unpack' in str(err)` seems unnecessarily fragile. A future Python version could change the exception message.

Would it be better to use a test like `if '\t' in line` instead of relying on ValueErrors to catch the problematic case?

Reply:
Yes, should have thought about those points! Now added the if condition